File: app/src/main/java/com/softhink/single/survey/InterestsFragment.java
package com.softhink.single.survey;


import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;

import com.igalata.bubblepicker.BubblePickerListener;
import com.igalata.bubblepicker.adapter.BubblePickerAdapter;
import com.igalata.bubblepicker.model.PickerItem;
import com.igalata.bubblepicker.rendering.BubblePicker;
import com.softhink.single.R;

import org.jetbrains.annotations.NotNull;

/**
 * A simple {@link Fragment} subclass.
 */
public class InterestsFragment extends Fragment
        implements BubblePickerListener, View.OnClickListener {

    private BubblePicker picker;
    private ImageButton btnNext;
    private ImageButton btnPrev;

    public InterestsFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_interests, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        picker = view.findViewById(R.id.picker);

        picker.setAdapter(new BubblePickerAdapter() {
            String[] titles = new String[]{
                    "Encontrar pareja",
                    "Diversión",
                    "Sexo casual",
                    "Conocer gente nueva"
            };

            @Override
            public int getTotalCount() {
                return titles.length;
            }

            @NotNull
            @Override
            public PickerItem getItem(int i) {
                PickerItem item = new PickerItem();
                item.setTitle(titles[i]);
                item.setColor(Color.parseColor("#DA7C86"));
                item.setTextColor(Color.WHITE);
                item.setBackgroundImage(ContextCompat.getDrawable(getContext(), R.drawable.background_bubble));
                return item;
            }
        });

        picker.setCenterImmediately(true);
        picker.setBubbleSize(30);
        picker.setListener(this);

        btnNext = view.findViewById(R.id.btnLastPage);
        btnPrev = view.findViewById(R.id.btnFirstPage);

        btnNext.setOnClickListener(this);
        btnPrev.setOnClickListener(this);
    }

    @Override
    public void onResume() {
        super.onResume();
        picker.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
        picker.onPause();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btnLastPage:
                getFragmentManager().beginTransaction().replace(R.id.containerPreferences,
                        new TastesFragment(),
                        TastesFragment.class.getSimpleName())
                        .addToBackStack(TastesFragment.class.getSimpleName())
                        .commit();
                break;

            case R.id.btnFirstPage:
                getFragmentManager().popBackStack();
                break;
        }
    }

    @Override
    public void onBubbleSelected(PickerItem pickerItem) {
        Log.i("SingleApp", "Burbuja seleccionada: " + pickerItem.getTitle());
    }

    @Override
    public void onBubbleDeselected(PickerItem pickerItem) {
        Log.i("SingleApp", "Burbuja desseleccionada: " + pickerItem.getTitle());
    }
}

File: app/src/main/java/com/softhink/single/survey/InterestsFragment.kt
package com.softhink.single.survey


import android.graphics.Color
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.core.content.ContextCompat
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.igalata.bubblepicker.BubblePickerListener
import com.igalata.bubblepicker.adapter.BubblePickerAdapter
import com.igalata.bubblepicker.model.PickerItem
import com.softhink.single.R
import kotlinx.android.synthetic.main.arrow_back_and_next.*
import kotlinx.android.synthetic.main.fragment_interests.*

/**
 * A simple [Fragment] subclass.
 */
class InterestsFragment : Fragment(), BubblePickerListener, View.OnClickListener {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_interests, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (savedInstanceState == null) {
            picker.adapter = object : BubblePickerAdapter {
                var titles = arrayOf("Encontrar pareja", "Diversión", "Sexo casual", "Conocer gente nueva")

                override val totalCount: Int
                    get() = titles.size

                override fun getItem(position: Int): PickerItem {
                    return PickerItem().apply {
                        title = titles[position]
                        color = Color.parseColor("#DA7C86")
                        textColor = Color.WHITE
                        backgroundImage = ContextCompat.getDrawable(context!!, R.drawable.bg_bubble_red)
                    }
                }
            }

            picker.centerImmediately = true
            picker.bubbleSize = 30
            picker.listener = this

            btnLastPage.setOnClickListener(this)
            btnFirstPage.setOnClickListener(this)
        }
    }

    override fun onResume() {
        super.onResume()
        picker!!.onResume()
    }

    override fun onPause() {
        super.onPause()
        picker!!.onPause()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnLastPage -> fragmentManager?.beginTransaction()?.
                    add(R.id.containerPreferences, TastesFragment(), TastesFragment::class.java.simpleName)?.
                    addToBackStack(TastesFragment::class.java.simpleName)
                    ?.commit()

            R.id.btnFirstPage -> fragmentManager?.popBackStack()
        }
    }

    override fun onBubbleSelected(pickerItem: PickerItem) {
        Log.i("SingleApp", "Burbuja seleccionada: " + pickerItem.title!!)
    }

    override fun onBubbleDeselected(pickerItem: PickerItem) {
        Log.i("SingleApp", "Burbuja desseleccionada: " + pickerItem.title!!)
    }
}